pkg/manager/member/tidb_member_manager.go

```diff
@@ -241,10 +249,103 @@ func (m *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.TidbC
 			return err
 		}
 	}
+	// set random password
+	if tc.Spec.TiDB.Initializer.CreatePassword && !tc.Status.TiDB.InitPasswordPhase && !tc.IsTLSClusterEnabled() {
```

Comment:
Let's define a new function `syncInitializer()` and add it to the `Sync` function before `syncTiDBStatefulSetForTidbCluster`.
We have to check the endpoint of the TiDB service before creating the password.
Refer to https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_member_manager.go#L321

Reply:
If there is any endpoint, no need to check the Pod ready below, right?